app/views/components/_autocomplete.html.erb

```diff
@@ -0,0 +1,51 @@
+<%
+  id ||= "autocomplete-#{SecureRandom.hex(4)}"
+  type ||= nil
+  error_id = "error-#{SecureRandom.hex(4)}"
+  error_items ||= []
+  width ||= nil
+  data_attributes ||= {}
+  input ||= {}
+  name ||= nil
+  label = label.symbolize_keys if label
+  aria = error_id if error_items.any?
+
+  root_classes = %w(app-c-autocomplete govuk-form-group)
+  root_classes << "app-c-autocomplete--#{width}" if width
+  root_classes << "govuk-form-group--error" if error_items.any?
+
+  if name && label
```

Comment:
What is this line doing?

Reply:
Only rendering an input if both `name` and `label` have been provided. (Input has zero value if it doesn't have a name so we can hook into it, and has very poor accessibility if there is no corresponding label).

Reply:
Sorry, I missed that this is wrapping the whole thing.

Reply:
This is not a blocker, but I'd consider moving this to it's own block for readability.

Reply:
By that, you mean doing...?

```erb
%>
<% if name && label %>
```

Agreed it would made the code more readable, but it also changes behaviour slightly (if name or label not provided, it should render nothing, but would now render "\n"). I could `.strip` in the tests to get it to pass, but it feels more correct to not render anything at all.